doc/manual/manual.tex

```diff
@@ -10337,43 +10337,76 @@ \subsubsection{Donea \& Huerta 2D box geometry benchmark}
+bug fixes available in the development version. Also these modifications make
+your results less reproducible. If you used your results in a publication, you
+could no longer say \textit{which} version of \aspect{} was used to produce
+these results, because you modified it yourself. Therefore, we discourage this
+form of modification for productive use (it can still be helpful for teaching).
+
+\item Create a feature branch: If you are familiar with the version control
+system \texttt{git} that we use to organize the development of \aspect{} (an
+excellent tutorial is available at:
+\url{http://swcarpentry.github.io/git-novice/}) you might think of creating a
+separate branch inside your \aspect{} repository and making your changes in
+this branch. This way you keep the history of your local modifications separate
+from the changes made to the main version. You can also uniquely describe the
+\aspect{} version you used for a set of models, and you can upload your branch
+to make your changes reproducible. This approach is also the ideal starting
+point if you intend to contribute your changes back, as it already is the first
+step of our guide to contributing back (see also
+Section~\ref{sec:contributing}).  However, for projects with functionality that
+is not intended to be merged into the main version (e.g. because it is too
+specific to be of general use) we have found that this approach is not ideal,
+as you will still run into conflicts when you want to update your \aspect{}
+version, and you need to merge the main version into your branch, or rebase the
+branch every time you want to update. Thus, while ideal for contributing to
+\aspect{} we do not recommend this approach for keeping model-specific
+functionality around.
+
+\item Create a shared library than contains your changes: The main benefit of
+the plugin architecture described in the paragraph above is that if you want to
+extend \aspect{} for your own purposes, you can do this in a separate set of
+files that describe your situation, rather than by modifying the \aspect{}
+source files themselves. This is advantageous, because (i) it makes it possible
+for you to update \aspect{} itself to a newer version without losing the
+functionality you added (because you did not make any changes to the \aspect{}
+files themselves), (ii) because it makes it possible to keep unrelated changes
+separate in your own set of files, in a place where they are simple to find,
+and (iii) because it makes it much easier for you to share your modifications
+and additions with others, you can for example include them as supplementary
+material in your publications. Of course you can (and should) also use version
+control on your separate set of files to keep track of which version of files
+was used for a given set of models. We will discuss the concept begind plugins
+in Section~\ref{sec:plugins}, and how to write a plugin in Section~{sec:write-plugin}.
```

Comment:
maybe link to an example that uses a shared library like cookbooks/prescribed_velocity/ ?